lib/style_parser/namespacer/postfixer.js

```diff
@@ -12,26 +12,21 @@ var css = require('css');
-  $styles_prefixed = str_replace('}', "}\n", $styles_prefixed);
-  */
-
-  //var postfixed = styles.replace(/(,?\h+|\h+(-?[_a-z]+[_a-z0-9-]*)+\h*|\n)(a|abbr|address|area|article|aside|audio|b|base|bdo|blockquote|body|br|button|canvas|caption|cite|code|col|colgroup|command|datalist|dd|del|details|dfn|div|dl|dt|em|embed|eventsource|fieldset|figcaption|figure|footer|form|h1|h2|h3|h4|h6|head|header|hgroup|hr|i|img|input|ins|kbd|keygen|label|legend|li|link|mark|map|menu|meta|meter|nav|ol|optgroup|option|output|p|param|pre|progress|q|ruby|rp|rt|samp|section|select|small|source|span|strong|sub|summary|sup|table|tbody|td|textarea|tfoot|th|thead|time|title|tr|ul|var|video|wbr)(\h*,|:[a-z-\(\)]+|\h*{)/ig, '$1' + postfix + '$2');
-  var postfixed = styles.replace(/(a|abbr|address|area|article|aside|audio|b|base|bdo|blockquote|body|br|button|canvas|caption|cite|code|col|colgroup|command|datalist|dd|del|details|dfn|div|dl|dt|em|embed|eventsource|fieldset|figcaption|figure|footer|form|h1|h2|h3|h4|h6|head|header|hgroup|hr|i|img|input|ins|kbd|keygen|label|legend|li|link|mark|map|menu|meta|meter|nav|ol|optgroup|option|output|p|param|pre|progress|q|ruby|rp|rt|samp|section|select|small|source|span|strong|sub|summary|sup|table|tbody|td|textarea|tfoot|th|thead|time|title|tr|ul|var|video|wbr)(\s*,|:[a-z-\(\)]+|\s*{)/ig, '$1' + postfix + '$2');
-/*
-  // add postfix to add tags without a class/id
-  $styles_prefixed = "\n".$styles_prefixed;
-  $regex = '/(,?\h+|\h+(-?[_a-z]+[_a-z0-9-]*)+\h*|\n)'; // match any comma, space, class name, or newline (before, so we know it's in the selector part of the css file)
-
-  $regex .= '(a|abbr|address|area|article|aside|audio|b|base|bdo|blockquote|body|br|button|canvas|caption|cite|code|col|colgroup|command|datalist|dd|del|details|dfn|div|dl|dt|em|embed|eventsource|fieldset|figcaption|figure|footer|form|h1|h2|h3|h4|h6|head|header|hgroup|hr|i|img|input|ins|kbd|keygen|label|legend|li|link|mark|map|menu|meta|meter|nav|ol|optgroup|option|output|p|param|pre|progress|q|ruby|rp|rt|samp|section|select|small|source|span|strong|sub|summary|sup|table|tbody|td|textarea|tfoot|th|thead|time|title|tr|ul|var|video|wbr)'; // match any valid tag
-  $regex .= '(\h*,|:[a-z-\(\)]+|\h*{)/iu'; // match spaces, or another, or a pseudo selector, or {
-  $styles_prefixed = preg_replace($regex, '$1$3.'.Sanitizer::CSS_POSTFIX.'$4', $styles_prefixed);
-  $styles_prefixed = trim($styles_prefixed); // remove leading new line
-
-  return $styles_prefixed;
-  */
-  cb(null, postfixed);
+  var obj = css.parse(styles);
+  var sheet = obj.stylesheet;
+  sheet.rules.forEach(function(rule, ruleIndex) {
+    if (rule.type === 'rule') {
+      rule.selectors.forEach(function(selector, selectorIndex) {
+        if (selector.indexOf(':') > -1) {
+          var selectorHash = selector.split(":");
+          obj.stylesheet.rules[ruleIndex].selectors[selectorIndex] = selectorHash[0] + postfix + ':' + selectorHash[1];
```

Comment:
In cases like this, I would move the complex expression on the left hand side of the assignment below, outside the if/else, then in the if/else only make the decision of what should be on the right hand side.